phoenix-core/src/main/java/org/apache/phoenix/index/GlobalIndexChecker.java

```diff
@@ -356,11 +362,23 @@ private boolean verifyRowAndRepairIfNecessary(List<Cell> cellList) throws IOExce
             if (verifyRowAndRemoveEmptyColumn(cellList)) {
                 return true;
             } else {
+                long repairStart = EnvironmentEdgeManager.currentTimeMillis();
+
                 byte[] rowKey = new byte[cell.getRowLength()];
                 System.arraycopy(cell.getRowArray(), cell.getRowOffset(), rowKey, 0, cell.getRowLength());
                 long ts = getMaxTimestamp(cellList);
                 cellList.clear();
-                repairIndexRows(rowKey, ts, cellList);
+
+                try {
+                    repairIndexRows(rowKey, ts, cellList);
+                    metricsSource.incrementIndexRepairs();
+                    metricsSource.updateIndexRepairTime(EnvironmentEdgeManager.currentTimeMillis() - repairStart);
```

Comment:
shall we update the repair time before incrementing the index repairs?

Reply:
Shouldn't matter. The above increment is likely a single instruction and would take a few thousandths of micro-sec to impact our next ms level metric. If that is not the case, we should be very sad that our metric collection framework itself skews the metrics :(